src/block/Furnace.php

```diff
@@ -90,4 +94,83 @@ public function onScheduledUpdate() : void{
 			$world->scheduleDelayedBlockUpdate($this->position, 1); //TODO: check this
 		}
 	}
+
+	public function doHopperPull(Hopper $hopperBlock) : bool{
+		$currentTile = $this->position->getWorld()->getTile($this->position);
+		if(!$currentTile instanceof TileFurnace){
+			return false;
+		}
+
+		$tileHopper = $this->position->getWorld()->getTile($hopperBlock->position);
+		if(!$tileHopper instanceof TileHopper){
+			return false;
+		}
+
+		$sourceInventory = $currentTile->getInventory();
+		$targetInventory = $tileHopper->getInventory();
+
+		$itemStack = $sourceInventory->getResult();
+
+		if($itemStack->isNull()){
+			return false;
+		}
+
+		$singleItem = $itemStack->pop();
+
+		$sourceInventory->removeItem($singleItem);
+		$targetInventory->addItem($singleItem);
+
+		return true;
+	}
+
+	public function doHopperPush(Hopper $hopperBlock) : bool{
```

Comment:
Maybe it's better to keep the same order for `doHopper...` as in the interface? So push and pull